pkg/engine/update.go

```diff
@@ -474,14 +475,22 @@ func newUpdateSource(ctx context.Context,
 	}
 
 	// If that succeeded, create a new source that will perform interpretation of the compiled program.
-	return deploy.NewEvalSource(plugctx, &deploy.EvalRunInfo{
-		Proj:        proj,
-		Pwd:         pwd,
-		Program:     main,
-		ProjectRoot: projectRoot,
-		Args:        args,
-		Target:      target,
-	}, defaultProviderVersions, dryRun), nil
+	schemaLoader := schema.NewPluginLoader(plugctx.Host)
```

Comment:
Does this do any caching or do we reload the schema each time we get a resource for it?

Reply:
Indeed -- caching is necessary for this to even work. I've fix this and added comments and updated the PR description/commit message to (hopefully) explain it all.